openghg/standardise/surface/_icos_corso.py

```diff
@@ -0,0 +1,443 @@
+        inlet_height_fname = filepath.name.split("_")[-4]
+
+        species, site, inlet = initial_checks_and_setup(
+            species=species_fname,
+            input_site=input_site,
+            site_fname=site_fname,
+            input_inlet=inlet,
+            inlet_height_fname=inlet_height_fname,
+        )
+
+        header, df = convert_icos_file_to_dataframe(filepath=filepath, rename_dict=rename_dict)
+
+        df["time"] = pd.to_datetime(df[["year", "month", "day", "hour", "minute"]])
+        df.index = df["time"]
+        df["sampling_start_date"] = pd.to_datetime(df[["year", "month", "day"]])
+
+        columns_to_keep = [
+            species,
+            "time",
+            "sampling_start_date",
+            "integration_time",
+            "weighted_std_err",
+            species + "_variability",
+            "sampling_pattern",
+            "flag",
+        ]
+
+        df = df = df[columns_to_keep]
+
+    df = clean_dataframe(df=df, species_name=species)
+    data = df.to_xarray()
+    data["flag"] = data["flag"].astype(str)
+
+    if sampling_period is None and f"{species}_sampling_period" in data.data_vars:
+        rounded_values = np.round(data[f"{species}_sampling_period"].values, decimals=2)
+        unique_values = np.unique(rounded_values)
+        if [3600, 3601] in unique_values and len([3600, 3601]) == len(unique_values):
+            sampling_period = "3600.0"
+
+    metadata = {
+        "site": site,
+        "species": species,
+        "inlet": inlet_height_fname,
+        "inlet_height_magl": inlet,
+        "sampling_period": sampling_period,
+        "network": network,
+        "instrument": instrument,
+    }
+
+    attributes = {"inlet_height_magl": metadata["inlet_height_magl"], "data_owner": "See data_owner_email"}
+
+    if measurement_type is not None:
+        metadata["measurement_type"] = measurement_type
+
+    f_header = [s for s in header if "MEASUREMENT UNIT" in s]
+    if len(f_header) == 1:
+        units = f_header[0].split(":")[1].lower().strip()
+        metadata["units"] = units
+    else:
+        raise ValueError("No unique MEASUREMENT UNIT in file header")
+
+    f_header = [s for s in header if "CONTACT POINT" in s]
+    if len(f_header) == 1:
+        data_owner_email = f_header[0].split(":")[1].strip()
+        metadata["data_owner_email"] = data_owner_email
+    else:
+        f_header = [s for s in header if "CONTACT POINT EMAIL" in s]
+        if len(f_header) == 1:
+            data_owner_email = f_header[0].split(":")[1].strip()
+            metadata["data_owner_email"] = data_owner_email
+        else:
+            raise ValueError("Couldn't identify data owner email")
+
+    if sampling_period is None:
+        f_header = [s for s in header if "TIME INTERVAL" in s]
+        interval_str = f_header[0].split(":")[1].strip()
+        if interval_str == "hourly":
+            metadata["sampling_period"] = "3600.0"
+        else:
```

Comment:
I liked the way you were doing this previously where you were checked for the words "integrated sampling" before setting the "sampling_period" value. We could have this as an if-elif-else block where for the else we raise an error and ask the user to specify a sampling_period directly.

Reply:
To clarify on this, I was suggesting checking "integrated sampling" was written in the header rather than including an `else:`
```suggestion
        elif "integrated sampling" in interval_str:
```